### ISSUE TITLE
cmd/vendor/github.com/google/pprof/internal/driver: TestHttpsInsecure flaky under load

### ISSUE THREAD

**Issue by josharian** [golang/go#24611]
This was also reported as #22594, where it was assumed to be specific to some builders. But the issue is that the test is flaky when the underlying machine is heavily loaded.

I can reproduce it on my speedy darwin/amd64 laptop with:

```
$ cd cmd/vendor/github.com/google/pprof/internal/driver
$ go test -c
$ stress -p 64 ./driver.test
```

Result: "849 runs so far, 24 failures", which is > 2% failure rate.

I also saw this happen during a regular local all.bash, and also earlier on a trybot run--which confused me into wasting time, assuming I had introduced a bug with a compiler change.

Also, the test takes 10-15s, 5s of which is spent burning CPU. :(

We should figure out how to make this non-flaky, and ideally also less resource-intensive. Or at the very least we should skip it in testing.Short mode, and leave it to be run once #12508 is fixed.

cc @hyangah (?)

Marking as Go 1.11 because I am frustrated with the number of flaky tests we have.

**Comment by mvdan** [comment]
@josharian what's the exact failure message you get? I wonder if this is https://github.com/google/pprof/issues/328.

Also see https://github.com/google/pprof/issues/343 - the pprof people are aware of the number of flakes they tend to introduce given our vast variety of builders.

**Comment by aalexand** [comment]
@josharian Curious - what's the stress tool you used? https://people.seas.harvard.edu/~apw/stress doesn't seem to have the -p flag.

**Comment by mvdan** [comment]
If it's a tool people in the Go team use, and it's not part of the Go tree, it's very usually in the x repos: https://godoc.org/golang.org/x/tools/cmd/stress

**Comment by josharian** [comment]
What is the procedure for re-vendoring? I'm excited to get the upsteam fix pulled in.

**Comment by mvdan** [comment]
@josharian last time I just did a copy manually, careful to not introduce unwanted parts of the repository. I wondered if a tool was being used to keep track of vendored packages, but apparently it's never been the case.

**Comment by mvdan** [comment]
I'll submit a CL now, since I was also wanting to re-vendor to pull in flakiness fixes: https://github.com/golang/go/issues/24405

**Comment by mvdan** [comment]
Actually, we are still blocked on an issue - #24508. I'll ping the pprof people in case they are going to have a fix soon, but otherwise I'll sync the vendored copy with that test disabled.
